client/landing/stepper/declarative-flow/new-hosted-site-flow.ts

```diff
@@ -55,6 +65,7 @@ const hosting: Flow = {
 
 		const query = useQuery();
 		const queryParams = Object.fromEntries( query );
+		const plan = queryParams.plan;
 		const flowName = this.name;
 
 		const goBack = () => {
```

Comment:
I am afraid we need to add a case for going back from `_currentStepSlug === 'plans' && query.has( 'partnerBundle' )` show that it goes back to domains step rather than in '/sites?hosting-flow=true'.

Reply:
That's right, well spotted. A similar logic that decides if the step must be executed needs to be applied to the `goBack` functionality, working on it.

Reply:
Refined the `goBack` functionality in [4b7b1ce](https://github.com/Automattic/wp-calypso/pull/95721/commits/4b7b1ce42df5ddb74079965b8a63654e1315dde0#diff-cc59aa46143159b6a6ca515435b7e71073c365342162deb1a2de7f43f1dee0fdR79)